content_scripts/storageLoader.js

```diff
@@ -334,17 +402,23 @@ async function scrapeWebsiteDOM(url, cssSelector, all = false) {
 async function getCourseSemester(courseId) {
     const ValidCourseIdLength = 5
     courseId = courseId.trim()
-    if (courseId.length > ValidCourseIdLength){
-        throw "course id invalid"
+    if (courseId.length > ValidCourseIdLength) {
+        courseId = courseId.substring(ValidCourseIdLength)
+        // throw "course id invalid"
     }
     // as syllabus expecting id with 5 chars we need to add prefix of 0s so the id will be with 5 chars
     if (courseId.length < ValidCourseIdLength) {
         let numOfZero = ValidCourseIdLength - courseId.length
         courseId = "0".repeat(numOfZero) + courseId
     }
+
     const semester = await scrapeWebsiteDOM(
         `https://shnaton.huji.ac.il/index.php/NewSyl/${courseId}/1/2021/`,
         '.hebItem:nth-of-type(4)')
+
```

Comment:
would like to hear your thoughts what to do if we couldn't find the semester

Reply:
You can add another 'semester' which its name is "undefined" and add the course to there.

Reply:
Hmm, I didn't like so much the idea of the "Unknown" section, and now the situation is that we just assign it to Semester A, I think it's not good as well.
There is an example of it in the course schedule when the location is not known and I think it's better not said.
Maybe this case in the course schedule is different from here and your right
let's wait and hear the other's opinions before changing it

Reply:
What are we doing with courses that are 
סמסטר א /או ב
?

Reply:
I think making a category for unclassified courses should be a solution. In my opinion the user will look for the course inside the category he is expecting to see it. So if a semester b course that we couldn't classify will be inside semester A, the user might not find it

Reply:
> What are we doing with courses that are
> סמסטר א /או ב
> ?

Do you have an example? for now, if it's on a and b and the syllabus shows סמסטר א' ו ב' it will consider it as semester b

Reply:
> I think making a category for unclassified courses should be a solution. In my opinion the user will look for the course inside the category he is expecting to see it. So if a semester b course that we couldn't classify will be inside semester A, the user might not find it

That makes sense for now I'll add Unknown category and put it last

Reply:
I've added the Semester Unknown.
Also, I've fixed the case that the dark mode ignores dimmed text.
@yanirelfasy @norbit8

Reply:
> > What are we doing with courses that are
> > סמסטר א /או ב
> > ?
> 
> Do you have an example? for now, if it's on a and b and the syllabus shows סמסטר א' ו ב' it will consider it as semester b

regarding that issue, maybe if it's both on semester A and B we can put it in Unknown?
Or we should default it to A instead of B?
What do you think?

Reply:
@idan2468 We can put it in A and in B, wdyt?

Reply:
Hmm I don't like it so much. The course list is already with many links. 
Besides, it's can lead to bugs with the other features like get course semester and removeCourses

Reply:
Yes that's true...
So right now they will be assigned to `Unkown`, right?

Reply:
> Yes that's true...
> So right now they will be assigned to `Unkown`, right?

Hmm for now it's considered B.
I though maybe change it to A because I think most of courses that are in A and B are like that because students register to A and if they fail they doing it on B no?

Reply:
Maybe just make a new section "A and B" and put it there?

Reply:
> Maybe just make a new section "A and B" and put it there?

It's possible... 
for now, I'll do it as you've said and we need to play with it and see if it's working fine.